apps/server/src/app/export.service.ts

```diff
@@ -1,7 +1,20 @@
-import { Injectable } from '@nestjs/common';
+import {Injectable} from '@nestjs/common';
 import { HtmlExporter } from './export/html';
 
 @Injectable()
 export class ExportService {
   public exports: HtmlExporter[] = [];
+
+  public cancel(packageId: string) {
+    const exporter = this.exports.find(e => e.packageId === packageId);
+    const index = this.exports.indexOf(exporter);
+
+    if (index >= 0) {
+      this.exports.splice(index, 1);
+      exporter.igPublisherProcess.kill("SIGKILL");
```

Comment:
Should there be a check here to make sure that the igPublisherProcess is still running?
Not sure if ChildProcess has a status code...
Example:
```if (exporter.igPublisherProcess && exporter.igPublisherProcess.status === 'running') {
  exporter.igPublisherProcess.kill('SIGKILL');
}```

Reply:
After some digging, I'm not sure that ChildProcess actually has a reasonable way to track whether it's currently running or not based on what I gathered about how it works.  From what I understand, as long as it exists, it's "running" and until it's gotten to the end of it's routine and has sat running while doing nothing for some system dependent period of time, it is then garbage collected by the system.  I think with the safeguards we have client side, i.e. not letting the user click cancel until the publisher is actually running and then immediately graying back out after being clicked, we probably don't need to worry about sending a kill signal to a process that's not running.  Let me know if you think differently and I'll try to find a reasonable solution.

Reply:
Agreed. Thanks!